### ISSUE TITLE
App crashing when library has an empty playlist

### ISSUE THREAD

**Issue by everypizza1** [Leonidius20/BeatInspector#3]
After logging in into my Spotify account, the app will crash while trying to load the home screen. I'm using the latest version from the IzzyOnDroid repo.

**Comment by Leonidius20** [comment]
Hello and thank you for reporting this issue.
I am unable to reproduce the issue at the moment. 
Pleae provide more information, including:
- When did the issue occur? (the api key was activated by Spotify only yesterday due to my mistake, and the key not being activated could be the cause of the crash. If you experienced the issue yesterday or earlier, please try again and check if anything changed)
- What is your Android version? 
-  What is the model of your device?
thank you.

**Comment by everypizza1** [comment]
It's still happening today, and I'm on Android 11 on a OnePlus Nord N10 5G. If it helps, I can access logcat.

**Comment by Leonidius20** [comment]
@everypizza1 
sorry for the delay.
Please install the new version (0.0.7) from the Releases section or from IzzyOnDroid when it becomes available there.
This new version improves crash reporting, specifically, when the app crashes, it shows a screen that offers you to copy crash data (stacktrace) into the clipboard.
Please copy the data from this screen and post it here. Alternatively, it this new crash screen fails to show up, please access logcat and post the stacktrace here. 
Thank you.

**Comment by everypizza1** [comment]
Here's the output from it:
```
Build version: 0.0.7 
Current date: 2024-06-16 09:19:47 
Device: OnePlus BE2028 
OS version: Android 11 (SDK 30) 
 
Stack trace:  
java.lang.NullPointerException: Parameter specified as non-null is null: method u8.s.d0, parameter <this>
	at u8.s.d0(SourceFile:3)
	at ua.leonidius.beatinspector.data.playlists.network.dto.SimplifiedPlaylistDto.getSmallestImageOrNull(SourceFile:3)
	at ua.leonidius.beatinspector.data.playlists.network.dto.SimplifiedPlaylistDto.toDomainObject(SourceFile:5)
	at gb.a$c.v(SourceFile:209)
	at gb.a$c.n(SourceFile:22)
	at v4.n$a.v(SourceFile:100)
	at v4.n$a.k(SourceFile:13)
	at ca.l.N(SourceFile:5)
	at ca.l.Q(SourceFile:74)
	at v4.m$a.v(SourceFile:86)
	at z8.a.j(SourceFile:9)
	at p9.n0.run(SourceFile:112)
	at p9.u0.G0(SourceFile:163)
	at ca.l.M(SourceFile:103)
	at v4.m.run(SourceFile:23)
	at c.p.run(SourceFile:237)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:923)
	Suppressed: u9.f: [w1{Cancelling}@aad233d, Dispatchers.Main.immediate]
```

It does seem to get further in the loading though.

**Comment by everypizza1** [comment]
I've figured out the problem: I somehow have an empty playlist. Deleting it fixed the issue.

**Comment by Leonidius20** [comment]
I see. I was able to reproduce the issue after creating an empty playlist in my account. The thing is, the app is not supposed to crash even with slightly out of the ordinary user data, so i will reopen the issue and hopefully will fix it in the next version.

**Comment by tuxnowar** [comment]
same issue here
this is the crash data

Build version: 0.0.7 
Current date: 2024-06-17 08:30:46 
Device: Xiaomi Redmi Note 9 Pro 
OS version: Android 12 (SDK 31) 
 
Stack trace:  
java.lang.NullPointerException: Parameter specified as non-null is null: method u8.s.d0, parameter <this>
	at u8.s.d0(SourceFile:3)
	at ua.leonidius.beatinspector.data.playlists.network.dto.SimplifiedPlaylistDto.getSmallestImageOrNull(SourceFile:3)
	at ua.leonidius.beatinspector.data.playlists.network.dto.SimplifiedPlaylistDto.toDomainObject(SourceFile:5)
	at gb.a$c.v(SourceFile:209)
	at gb.a$c.n(SourceFile:22)
	at v4.n$a.v(SourceFile:100)
	at v4.n$a.k(SourceFile:13)
	at ca.l.N(SourceFile:5)
	at ca.l.Q(SourceFile:74)
	at v4.m$a.v(SourceFile:86)
	at z8.a.j(SourceFile:9)
	at p9.n0.run(SourceFile:112)
	at p9.u0.G0(SourceFile:163)
	at ca.l.M(SourceFile:103)
	at v4.m.run(SourceFile:23)
	at c.p.run(SourceFile:237)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:920)
	Suppressed: u9.f: [w1{Cancelling}@4dd0f96, Dispatchers.Main.immediate]

**Comment by Nosen92** [comment]
I seem to have this issue too. I also have an empty playlist in my library. Here is my log copy-paste:
Build version: 0.0.7 
Current date: 2024-06-17 15:34:38 
Device: OnePlus ONEPLUS A6003 
OS version: Android 14 (SDK 34) 
 
Stack trace:  
java.lang.NullPointerException: Parameter specified as non-null is null: method u8.s.d0, parameter <this>
	at u8.s.d0(SourceFile:3)
	at ua.leonidius.beatinspector.data.playlists.network.dto.SimplifiedPlaylistDto.getSmallestImageOrNull(SourceFile:3)
	at ua.leonidius.beatinspector.data.playlists.network.dto.SimplifiedPlaylistDto.toDomainObject(SourceFile:5)
	at gb.a$c.v(SourceFile:209)
	at gb.a$c.n(SourceFile:22)
	at v4.n$a.v(SourceFile:100)
	at v4.n$a.k(SourceFile:13)
	at ca.l.N(SourceFile:5)
	at ca.l.Q(SourceFile:74)
	at v4.m$a.v(SourceFile:86)
	at z8.a.j(SourceFile:9)
	at p9.n0.run(SourceFile:112)
	at p9.u0.G0(SourceFile:163)
	at ca.l.M(SourceFile:103)
	at v4.m.run(SourceFile:23)
	at c.p.run(SourceFile:237)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
	Suppressed: u9.f: [w1{Cancelling}@66b7d67, Dispatchers.Main.immediate]

**Comment by Leonidius20** [comment]
Should be fixed in version 0.0.8, which is already available in the Releases section here and will be picked up by IzzyOnDroid soon.

**Comment by Nosen92** [comment]
Can confirm that this fixed the issue!
